db/functions/redundant.py

```diff
@@ -4,18 +4,33 @@
 Mathesar filters not supporting composition.
 """
 
-from db.functions.base import DBFunction
+from abc import abstractmethod
+
+from db.functions.base import DBFunction, Or, Lesser, Equal, Greater
 
 from db.functions import hints
 
 
-# TODO (tech debt) define in terms of other DBFunctions
-# would involve creating an alternative to to_sa_expression: something like to_db_function
-# execution engine would see that to_sa_expression is not implemented, and it would look for
-# to_db_function.
+class RedundantDBFunction(DBFunction):
+    """
+    A DBFunction that is meant to be unpacked into another DBFunction. A way to define a DBFunction
+    as a combination of DBFunctions. Its to_sa_expression method is not to used. Its concrete
+    implementations are expected to implement the unpack method.
+    """
+    @staticmethod
+    def to_sa_expression(*_):
+        raise Exception("UnpackabelDBFunction.to_sa_expression should never be used.")
+
+    @abstractmethod
+    def unpack(self):
```

Comment:
For my own education:  Should implementations of this `unpack` method recurse "in place"?  I.e., if I happen to have a redundant (or whatever we decide for a term) function that's composed of other redundant functions, should the `unpack` method call the unpack methods of those functions as well, or should that be left to the caller?  I'm not sure since you'd want to avoid having the recursion in multiple places, and `_db_function_to_sa_expression` is already recursive.

Reply:
I think it's desirable not to have to do manual recursion inside `unpack`. I don't immediately recall if I implemented full recursion yet. I think I did, since I was concerned about infinite loops.